### PR TITLE
make it so upgrade connections can be built from the outside

### PR DESCRIPTION
**Pull request by jbr**

### CODE REVIEW

File: src/upgrade/connection.rs [hunk 1]
```diff
@@ -9,6 +9,12 @@ pub struct RawConnection<Inner> {
     inner: Inner,
```
Comment (Fishrock123) [inline src/upgrade/connection.rs:9]: If `Connection` is `RawConnection<Box<dyn T>>` anyways, why isn't `inner just directly `Box<dyn T>`?

**Review by Fishrock123 — COMMENTED** [review]
This structure strikes me as being a bit odd, see comment below.

However, along the lines of this PR, might we also want a `From<Box<dyn T>>`?

**Comment by jbr** [comment]
This will only ever be used in one place probably, so it's not worth adding an Into. I think this code probably could be simplified, but that's not what this PR is about

**Review by Fishrock123 — APPROVED** [review]

File: src/upgrade/connection.rs (hunk 1 above)
Comment (yoshuawuyts) [reply src/upgrade/connection.rs:9]: Good question; not sure. I think we should pick that up when stabilizing the `upgrade`submodule.

**Review by yoshuawuyts — APPROVED** [review]

**Pull request merged**